src/Illuminate/Foundation/Testing/DatabaseTransactions.php

```diff
@@ -4,12 +4,42 @@
 
 trait DatabaseTransactions
 {
+    /**
+     * The connections currently affected by the transaction. Indexed by connection name.
+     *
+     * @var array
+     */
+    protected $transactingConnections = [];
+
+    /**
+     * Retrieve the names of the database connections that should be included in the transaction.
+     *
+     * The default value of empty string will cause only the default database connection to be included in the transaction.
+     *
+     * @return array
+     */
+    protected function connectionsToTransact()
+    {
+        return [''];
+    }
+
     public function beginDatabaseTransaction()
     {
-        $this->app->make('db')->beginTransaction();
+        $db = $this->app->make('db');
+
+        foreach ($this->connectionsToTransact() as $connectionName) {
+            $connection = $db->connection($connectionName);
+            $this->transactingConnections[$connectionName] = $connection;
+
+            $connection->beginTransaction();
+        }
 
         $this->beforeApplicationDestroyed(function () {
-            $this->app->make('db')->rollBack();
+
+            foreach ($this->connectionsToTransact() as $connectionName) {
+                $this->app->make('db')->connection($connectionName)->rollBack();
+            }
+
```

Comment:
> Since I'm supposed to "look at every other file in the framework

That is correct. That's the most complete documentation of our code style. It we were to write it out formally, it would be massive and impossible to read.